doc/releases/changelog-dev.md

```diff
@@ -246,6 +220,47 @@
   >>> DeviceArray([[0.0, 0.0, 0.3650435], [0.0, 0.0, -0.3650435]], dtype=float32)
   ```
 
+* The kernel matrix utility functions in `qml.kernels` are now autodifferentiation-compatible.
+  In addition they support batching, for example for quantum kernel execution with shot vectors.
+  [(#3742)](https://github.com/PennyLaneAI/pennylane/pull/3742)
+
+  In addition to the autodifferentiation support in JAX, Autograd, Tensorflow and PyTorch,
+  optional batching was added, allowing for the following:
+  
+  ```python
+  dev = qml.device('default.qubit', wires=2, shots=(100, 100))
+  @qml.qnode(dev)
+  def circuit(x1, x2):
+      qml.templates.AngleEmbedding(x1, wires=dev.wires)
+      qml.adjoint(qml.templates.AngleEmbedding)(x2, wires=dev.wires)
+      return qml.probs(wires=dev.wires)
+
+  kernel = lambda x1, x2: circuit(x1, x2)[:, 0]
+  ```
+
+  Note that we extract the first probability vector entry for both
+  evaluations using 100 shots each.
+  We can then compute the kernel matrix on a set of 4 (random) feature
+  vectors ``X`` but using two sets of 100 shots each via
+
+  ```pycon
+  >>> X = np.random.random((4, 2))
+  >>> qml.kernels.square_kernel_matrix(X, kernel)
+  tensor([[[1.  , 0.86, 0.88, 0.92],
+           [0.86, 1.  , 0.75, 0.97],
+           [0.88, 0.75, 1.  , 0.91],
+           [0.92, 0.97, 0.91, 1.  ]],
+
+          [[1.  , 0.93, 0.91, 0.92],
+           [0.93, 1.  , 0.8 , 1.  ],
+           [0.91, 0.8 , 1.  , 0.91],
+           [0.92, 1.  , 0.91, 1.  ]]], requires_grad=True)
+  ```
+
+<h4>Smartly decompose Hamiltonian evolution 💯</h4>
+
+To be merged this week
+
 <h4>Tools for quantum chemistry and other applications 🛠️</h4>
 
 * A new method called `qml.qchem.givens_decomposition` has been added, which decomposes a unitary into a sequence
```

Comment:
> In the next line, line 267, can we just say
>
> >  of Givens rotation gates.
>
> ?

I'm not sure, looking [here](https://docs.pennylane.ai/en/latest/code/api/pennylane.qchem.givens_decomposition.html?highlight=givens) I would read the sentence as 

> Decompose a unitary into (a sequence of Givens rotation gates with phase shifts) and (a diagonal phase matrix.)

i.e., the diagonal phase matrix separate from Givens rotations. Perhaps some rewording could help?
> decomposes a unitary into a sequence of Givens rotation gates with phase shifts as well as a diagonal phase matrix.

Reply:
> Also, the proceeding code example can probably just all go in a pycon environment sweat_smile

:+1:  here's how it looks now

![image](https://user-images.githubusercontent.com/49409390/219677381-3d9046ff-0b39-489f-8b61-4d0497ad1fed.png)

Reply:
> Line 290 suggestion for consistency:
>
> > A new template called BasisRotation has been added, which performs a basis transformation defined by a set of
>
> @josh146 is the preferred convention here to prefix everything with `qml.`? :thinking:

I prefer we prefix everything with the correct path so that users can easily access simply using `import pennylane as qml`, however I don't think it's worth the effort to make the change this time :sweat_smile: